.github/workflows/build_and_test.yml

```
@@ -1092,12 +1093,46 @@ jobs:
               }
 
               console.log(`Release for tag ${tag} does not exist. Creating it...`);
+
+              // Get the merge commit message to extract the PR number
+              const commit = await github.rest.git.getCommit({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                commit_sha: context.sha,
+              });
+              const prMatch = commit.data.message.match(/Merge pull request #(\d+)/);
+              const prNumber = prMatch ? prMatch[1] : null;
+
+              // Find the previous release on the branch
+              const releases = await github.rest.repos.listReleases({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                per_page: 100,
+              });
+
+              const branchReleases = releases.data
+                .filter((release) => {
+                  const regex = new RegExp(`^${branch}-\\d+$`);
+                  return regex.test(tag) && !release.draft && !release.prerelease;
```

It seems you're testing tag that you just created, but need to test a tag from the release. Something like this:

```suggestion
                  return regex.test(release.tag_name) && !release.draft && !release.prerelease;
```

---

:facepalm: